### PR TITLE
Rewrites `GetExactComponent` to be more readable

### PR DESCRIPTION
**Pull request by DGamerL**
## About The Pull Request
Rewrites `GetExactComponent` to be 1641 compliant, and more readable
## Why It's Good For The Game
This proc was quite unreadable, and less work to make the server run on 1641
## Changelog
:cl:
refactor: refactored `GetExactComponent` to be 1641 compatible
/:cl:

### CODE REVIEW

**Comment by SyncIt21** [comment]
This isn't worthy of a refactor label but a code improvement at best.

**Comment by SyncIt21** [comment]
This isn't a refactor but a code improvement

File: code/datums/components/_component.dm [hunk 1]
```diff
@@ -273,19 +273,17 @@
...
 		return null
-	var/datum/component/C = dc[c_type]
-	if(C)
-		if(length(C))
-			var/list/component_list = C
-			C = component_list[1]
-		if(C.type == c_type)
-			return C
-	return null
+	var/datum/component/C
```
Comment (san7890) [inline code/datums/components/_component.dm:282]: change this from a single-letter component to a named component (e.g. `potential_component`)

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/datums/components/_component.dm [hunk 2]
```diff
@@ -273,19 +273,17 @@
...
+	if(length(all_components))
+		C = all_components[c_type]
+	if(C?.type == c_type)
+		return C
```
Comment (san7890) [inline code/datums/components/_component.dm:286]:
```suggestion
		return C
	
	return null
```

Comment (DGamerL) [reply code/datums/components/_component.dm]: Done!

File: code/datums/components/_component.dm (hunk 1 above)
Comment (DGamerL) [reply code/datums/components/_component.dm]: Done!

**Pull request merged**